src/scanner/mod.rs

```diff
@@ -130,8 +137,33 @@ impl Scanner {
     ///
     /// Note: `self` must contain `self.ip`.
     async fn scan_socket(&self, socket: SocketAddr) -> io::Result<SocketAddr> {
-        let tries = self.tries.get();
+        if self.sudp {
+            // we try different wait times and packet types
+            // I looked for a crate to create these packets but couldn't find one
+            let waits = vec![0, 50, 100, 300];
+            let payloads = vec![
+                craft_dns_query_packet(),
+                craft_ntp_packet(),
+                craft_dhcpc_packet(),
+            ];
 
+            for payload in &payloads {
+                for &wait_ms in &waits {
+                    let wait = Duration::from_millis(wait_ms);
+                    match self.udp_scan(socket, payload.clone(), wait).await {
+                        Ok(true) => return Ok(socket), // Successful scan
+                        Ok(false) => continue,         // Timed out, try next wait time or payload
+                        Err(e) => return Err(e),       // Some other error occurred
+                    }
+                }
+            }
+            return Err(io::Error::new(
+                io::ErrorKind::TimedOut,
+                "All payloads and wait times failed",
```

Comment:
I'm not so sure about this error? Is it attached to the If statement? I do not think TimedOut will be good here. And I am not sure we need an error here unless Rust is complaining :)